File: SwrveSDK/SDK/Messaging/SwrveMessageButtonDetails.m
#import "SwrveMessageButtonDetails.h"

@implementation SwrveMessageButtonDetails

@synthesize buttonName;
@synthesize buttonText;
@synthesize actionType;
@synthesize actionString;

- (id)initWith:(NSString *)name buttonText:(NSString *)text actionType:(SwrveActionType)type actionString:(NSString *)action {
    if (self = [super init]) {
        self.buttonName = name;
        self.buttonText = text;
        self.actionType = type;
        self.actionString = action;
    }
    return self;
}
@end

File: SwrveSDKSwift/SwrveMessageButtonDetails.swift
@objc public class SwrveMessageButtonDetails: NSObject {
    @objc public var buttonName: String
    @objc public var buttonText: String?
    @objc public var actionType: SwrveActionType
    @objc public var actionString: String

    @objc public init(with buttonName: String, buttonText: String?, actionType: SwrveActionType, actionString: String) {
        self.buttonName = buttonName
        self.buttonText = buttonText
        self.actionType = actionType
        self.actionString = actionString
        super.init()
    }
}
